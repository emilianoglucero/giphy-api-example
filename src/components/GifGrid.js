import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGriditem } from './GifGriditem';

export const GifGrid = ( { category } ) => {

    const { loading, data:images } = useFetchGifs( category );


    return (
        <>
        <h3 className="animate__fadeIn animate__flash">{ category }</h3>

        { loading && <p className="animate__flash">Loading...</p> }

        <div className="card-grid"> 
            {
                images.map( img => (
                    <GifGriditem 
                        title={ img.title }
                        url={ img.url }
                        key={img.id}
                    /> 
                ))
            }
        </div>
        </>
    )
}
