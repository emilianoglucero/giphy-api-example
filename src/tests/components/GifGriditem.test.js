import React from 'react';
import '../../setupTests';

import '@testing-library/jest-dom';
import { GifGriditem } from '../../components/GifGriditem';
import { shallow } from 'enzyme';



describe('Pruebas en el gifgriditem', () => {

    const title = "Goku";
    const url = "https://ghiphy.com"
    const wrapper = shallow( <GifGriditem title={ title } url={ url} /> );

    test('debe mostrar el componenete correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe tener un parrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim()).toBe( title );
        
    })

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        console.log(img.props());
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    })

    test('debe de tener el animate__fadeIn en el class', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');
        //animate fadein
        console.log(div.prop('className'));

        expect( className.includes('card animate__fadeIn') ).toBe( true );
        
    })
    
    
    


})
