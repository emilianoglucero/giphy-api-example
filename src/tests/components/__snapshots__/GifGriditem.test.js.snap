// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el gifgriditem debe mostrar el componenete correctamente 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="Goku"
    src="https://ghiphy.com"
  />
  <p>
     
    Goku
     
  </p>
</div>
`;
